File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Displaying Video Files Stored in Azure/C#/AZURE_DisplayVideoStoredInAzure/AZURE_DisplayVideoStoredInAzure/SilverlightWebPart/SilverlightWebPart.cs
using System;
using System.ComponentModel;
using System.Runtime.InteropServices;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using Microsoft.SharePoint;
using Microsoft.SharePoint.WebControls;
using System.Drawing;

namespace AZURE_DisplayVideoStoredInAzure.SilverlightWebPart
{
    [ToolboxItemAttribute(false)]
    public class SilverlightWebPart : WebPart
    {
        private SilverlightPluginGenerator _silverlightPluginGenerator = null;

        public SilverlightWebPart()
        {
            this._silverlightPluginGenerator = new SilverlightPluginGenerator
            {
                Source = new Uri("/SiteAssets/Silverlight/AZURE_SilverlightVideoApp/AZURE_SilverlightVideoApp.xap", UriKind.Relative),
                Width = new Unit(400, UnitType.Pixel),
                Height = new Unit(300, UnitType.Pixel),
                BackGround = Color.White,
                Version = SilverlightVersion.v3,
                AutoUpgrade = true,
                OnError = "onSilverlightError",
            };
        }

        protected override void CreateChildControls()
        {
            base.CreateChildControls();

            this.Controls.Add(new LiteralControl(@"<script type=""text/javascript"">" + Resources.onSilverlightErrorHandler + "</script>"));

            // Set the SiteUrl here.  Can't do this earlier since SPContext may be null during instantiation
            this._silverlightPluginGenerator.InitParams.Add(new InitParam("SiteUrl", SPContext.Current.Site.Url));
            this.Controls.Add(new LiteralControl(this._silverlightPluginGenerator.ToString()));
        }

        protected override void RenderContents(HtmlTextWriter writer)
        {
            base.RenderContents(writer);
        }
    }
}

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Displaying Video Files Stored in Azure/VB/AZURE_DisplayVideoStoredInAzure/AZURE_DisplayVideoStoredInAzure/SilverlightWebPart/SilverlightWebPart.vb
<ToolboxItemAttribute(False)> _
Public Class SilverlightWebPart
    Inherits WebPart
    Private _silverlightPluginGenerator As SilverlightPluginGenerator = Nothing

    Public Sub New()

        Me._silverlightPluginGenerator = New SilverlightPluginGenerator
        With Me._silverlightPluginGenerator
            .Source = New Uri("/SiteAssets/Silverlight/AZURE_SilverlightVideoApp/AZURE_SilverlightVideoApp.xap", UriKind.Relative)
            .Width = New Unit(400, UnitType.Pixel)
            .Height = New Unit(300, UnitType.Pixel)
            .BackGround = Drawing.Color.White
            .Version = SilverlightVersion.v3
            .AutoUpgrade = True
        End With

    End Sub

    Protected Overrides Sub CreateChildControls()
        MyBase.CreateChildControls()

        ' Set the SiteUrl here.  Can't do this earlier since SPContext may be null during instantiation
        Me._silverlightPluginGenerator.InitParams.Add(New InitParam("SiteUrl", SPContext.Current.Site.Url))
        Me.Controls.Add(New LiteralControl(Me._silverlightPluginGenerator.ToString()))
    End Sub

    Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
        MyBase.RenderContents(writer)
    End Sub
End Class
